src/rETHCurveUniv3Callee.sol

```diff
@@ -0,0 +1,192 @@
+    }
+    function _divup(uint256 x, uint256 y) internal pure returns (uint256 z) {
+        z = _add(x, _sub(y, 1)) / y;
+    }
+
+    constructor(address uniV3Router_, address daiJoin_, address weth_) public {
+        uniV3Router    = UniV3RouterLike(uniV3Router_);
+        daiJoin        = DaiJoinLike(daiJoin_);
+        TokenLike dai_ = DaiJoinLike(daiJoin_).dai();
+        dai            = dai_;
+        weth           = weth_;
+
+        dai_.approve(daiJoin_, type(uint256).max);
+    }
+
+    receive() external payable {}
+
+    function _fromWad(address gemJoin, uint256 wad) internal view returns (uint256 amt) {
+        amt = wad / 10 ** (_sub(18, GemJoinLike(gemJoin).dec()));
+    }
+
+    function clipperCall(
+        address sender,            // Clipper caller, pays back the loan
+        uint256 owe,               // Dai amount to pay back        [rad]
+        uint256 slice,             // Gem amount received           [wad]
+        bytes calldata data        // Extra data, see below
+    ) external {
+        (
+            address to,            // address to send remaining DAI to
+            address gemJoin,       // gemJoin adapter address
+            uint256 minProfit,     // minimum profit in DAI to make [wad]
+            bytes memory path,     // uniswap v3 path
+            address charterManager // pass address(0) if no manager
+        ) = abi.decode(data, (address, address, uint256, bytes, address));
+
+        address gem = GemJoinLike(gemJoin).gem(); // RocketPool rETH
+
+        // Convert slice to token precision
+        slice = _fromWad(gemJoin, slice);
+
+        // Exit gem to token
+        if(charterManager != address(0)) {
+            CharterManagerLike(charterManager).exit(gemJoin, address(this), slice);
+        } else {
+            GemJoinLike(gemJoin).exit(address(this), slice);
+        }
+
+        // rETH -> wstETH
+        TokenLike(gem).approve(rocketToLido, slice);
+        slice = CurvePoolLike(rocketToLido).exchange({
+            i:      0,     // send token id 1 (RocketPool rETH)
+            j:      1,     // receive token id 0 (wstETH)
+            dx:     slice, // send `slice` amount of rETH
+            min_dy: 0      // accept any amount of ETH (`minProfit` is checked below)
+        });
+        gem = CurvePoolLike(rocketToLido).coins(1);
+
+        // wstETH -> stETH
+        slice = WstEthLike(gem).unwrap(slice);
+        gem = WstEthLike(gem).stETH();
+
+        // stETH -> ETH
+        TokenLike(gem).approve(lidoToETH, slice);
+        slice = CurvePoolLike(lidoToETH).exchange({
+            i:      1,     // send token id 1 (stETH)
+            j:      0,     // receive token id 0 (ETH)
+            dx:     slice, // send `slice` amount of stETH
+            min_dy: 0      // accept any amount of ETH (`minProfit` is checked below)
+        });
+
+        // ETH -> wETH
+        WethLike(weth).deposit{
+            value: slice
+        }();
+        gem = weth;
```

Comment:
It was done on purpose to make it more understandable : the gem is now wETH because it was just wrapped.

However, I do see your point of making it as close as possible as WstETH. That also makes the code more understandable for those who compare the two files.

So if you insist, I’ll do what you suggest.

Reply:
Yeah, I think let's do it the same.

Reply:
done in https://github.com/makerdao/exchange-callees/pull/31/commits/17c783bfd4d89649365a6330994b4e2681ca4587